addons/web/static/src/core/l10n/localization_service.js

```diff
@@ -56,6 +61,10 @@ export const localizationService = {
                 }
             }
             Object.assign(translatedTerms, terms);
+            Object.assign(
+                translatedTermsGlobal,
+                Object.values(terms).reduce((acc, x) => ({ ...acc, ...x }), {})
+            );
```

Comment:
It's 1 line instead of 4 and way more readable to simply do:
```js
const terms = {};
for (const addon of Object.keys(result.modules)) {
    terms[addon] = {};
    for (const message of result.modules[addon].messages) {
        terms[addon][message.id] = message.string;
        translatedTermsGlobal[message.id] = message.string;
    }
}
```